src/SqlAsyncCollector.cs

```diff
@@ -95,7 +96,19 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
                 TelemetryInstance.TrackEvent(TelemetryEventName.AddAsync);
                 try
                 {
-                    this._rows.Add(item);
+                    if (typeof(T) == typeof(string))
+                    {
+                        this._rows.Add(JsonConvert.DeserializeObject<T>(item as string));
+                    }
+                    else if (typeof(T) == typeof(JObject))
+                    {
+                        dynamic itemAsJObject = JObject.Parse(item.ToString());
```

Comment:
So I realized that there's a more root problem - T is going to be whatever the type is originally (string, JObject, whatever), but then you're trying to change the type that you insert into the rows list. But the rows list is defined as `List<T>` which implies that it'll have the same type....so what you're trying to do here is breaking that contract. 

You need to be looking at either changing the type of _rows or not doing any conversion here and instead doing it later on somewhere.